webhook/icinga.go

```diff
@@ -165,6 +157,38 @@ func updateOrCreateService(icinga icinga2.Client,
 		serviceData.EnableActiveChecks = true
 	}
 
+	return serviceData
+}
+
+// updateOrCreateService updates or creates an Icinga2 service object from the
+// alert passed to the method
+func updateOrCreateService(icinga icinga2.Client,
+	hostname string,
+	serviceName string,
+	displayName string,
+	alert template.Alert,
+	c config.Configuration) (icinga2.Service, error) {
+
+	l := c.GetLogger()
+
+	// Check if this alert is a heartbeat alert and extract interval if so
+	heartbeat_interval := time.Duration(0)
+	if val, ok := alert.Labels["heartbeat"]; ok {
+		if alert.Status == "resolved" {
+			l.Infof("Not processing resolved heartbeat for %v", serviceName)
+			return icinga2.Service{}, nil
+		}
+		interval, err := time.ParseDuration(val)
```

Comment:
I couldn't figure out how to assign the parsed duration directly to `heartbeat_interval` in the outer scope here, thus the extra variable.

Reply:
Something is itching me but all I can come up is as good or bad as this. I guess we are on the edge of being in need for a builder. For now, this is overkill.